gcloud-java-dns/src/main/java/com/google/cloud/dns/testing/LocalDnsHelper.java

```diff
@@ -328,6 +361,62 @@ public void handle(HttpExchange exchange) throws IOException {
           requestMethod, exchange.getRequestURI())));
     }
 
+    private Response handleBatch(final HttpExchange exchange) throws IOException {
+      String contentType = exchange.getRequestHeaders().getFirst("Content-type");
+      if (contentType != null) {
+        int port = server.getAddress().getPort();
+        HttpMediaType httpMediaType = new HttpMediaType(contentType);
+        String boundary = httpMediaType.getParameter("boundary");
+        MultipartStream multipartStream =
+            new MultipartStream(exchange.getRequestBody(), boundary.getBytes(), 1024, null);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        byte[] bytes = new byte[1024];
+        multipartStream.skipPreamble();
+        while (multipartStream.readBoundary()) {
+          Socket socket = new Socket("localhost", port);
+          OutputStream socketOutput = socket.getOutputStream();
+          ByteArrayOutputStream section = new ByteArrayOutputStream();
+          multipartStream.readBodyData(section);
+          BufferedReader reader = new BufferedReader(
```

Comment:
Fine as is, though because this is basically treated as a String (converted from byte[] using local encoding - is that desired or should it be UTF-8 ?) you could have used many String specific tokenizing (using, Scanner, StringTokenizer, split,...).

e.g. 
```java
Scanner scanner = new Scanner(new String(section.toByteArray()));
while (scanner.hasNextLine()) {
  String line = scanner.nextLine();
  ...
}
```